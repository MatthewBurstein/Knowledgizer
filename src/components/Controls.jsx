import React, { Component } from 'react';

class Controls extends Component {
  render() {
    return(
      <div>
        <button
          id='clear'
          onClick={this.props.clearArticles}
        >
        Clear
      </button>
      </div>
    )
  }
}

export default Controls
