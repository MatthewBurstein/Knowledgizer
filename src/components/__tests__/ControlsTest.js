import React from 'react';
import { shallow, mount } from 'enzyme';
import Controls from '../Controls';

describe('Controls', () => {
  const mockClear = jest.fn()
  const mockStart = jest.fn()
  const mockStop = jest.fn()
  const props = { clearArticles: mockClear, startPrinting: mockStart, stopPrinting: mockStop }
  const controls = shallow(<Controls { ...props } />);

  it('renders correctly', () => {
    expect(controls).toMatchSnapshot();
  });

  describe('clearScreen button', () => {
    it('calls the clearScreen method from props', () => {
      controls.find('#clear').simulate('click')
      expect(mockClear).toHaveBeenCalled()
    })
  })

  describe('stop and start printing', () => {
    let controlsWhenPrinting, controlsWhenNotPrinting;
    beforeEach(() => {
      const propsWhenPrinting = { ...props, isPrinting: true }
      controlsWhenPrinting = mount(<Controls { ...propsWhenPrinting }/>)
      const propsWhenNotPrinting = { ...props, isPrinting: false }
      controlsWhenNotPrinting = mount(<Controls { ...propsWhenNotPrinting }/>)
      mockStart.mockClear()
      mockStop.mockClear()
    })

    describe('stop button', () => {
      it('when printing, it calls the stopPrinting method from props', () => {
        controlsWhenPrinting.find('#stop-printing').simulate('click')
        expect(mockStop).toHaveBeenCalled();
      })

      it('when not printing, is disabled', () => {
        controlsWhenNotPrinting.find('#stop-printing').simulate('click')
        expect(mockStop).not.toHaveBeenCalled();
      })
    })

    describe('start button', () => {
      it('when not printing, calls the startPrinting method from props', ()  => {
        controlsWhenNotPrinting.find('#start-printing').simulate('click');
        expect(mockStart).toHaveBeenCalled();
      })

      it('when printing, is disabled', () => {
        controlsWhenPrinting.find('#start-printing').simulate('click');
        expect(mockStart).not.toHaveBeenCalled();
      })
    })
  })
});
