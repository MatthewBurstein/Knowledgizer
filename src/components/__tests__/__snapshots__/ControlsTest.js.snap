// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Controls
    clearArticles={[Function]}
    startPrinting={[Function]}
    stopPrinting={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          id="clear"
          onClick={[Function]}
        >
          Clear
        </button>,
        <button
          disabled={true}
          id="stop-printing"
          onClick={[Function]}
        >
          Stop
        </button>,
        <button
          disabled={undefined}
          id="start-printing"
          onClick={[Function]}
        >
          Start
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Clear",
          "id": "clear",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Clear",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Stop",
          "disabled": true,
          "id": "stop-printing",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Stop",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Start",
          "disabled": undefined,
          "id": "start-printing",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Start",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            id="clear"
            onClick={[Function]}
          >
            Clear
          </button>,
          <button
            disabled={true}
            id="stop-printing"
            onClick={[Function]}
          >
            Stop
          </button>,
          <button
            disabled={undefined}
            id="start-printing"
            onClick={[Function]}
          >
            Start
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Clear",
            "id": "clear",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Clear",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Stop",
            "disabled": true,
            "id": "stop-printing",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Stop",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Start",
            "disabled": undefined,
            "id": "start-printing",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Start",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
